specs/AdaptiveCardsXamarinSpec.md

```diff
@@ -0,0 +1,188 @@
+
+    // -(NSString *)data;
+    [Export("data")]
+    string Data { get; }
+
+    // -(NSData *)additionalProperty;
+    [Export("additionalProperty")]
+    NSData AdditionalProperty { get; }
+}
+```
+
+#### Structs
+``` C#
+[Native]
+public enum ACRActionType : long
+{
+	ShowCard = 1,
+	Submit,
+	OpenUrl
+}
+```
+
+### APIs to be included in Xamarin
+
+| API Name | iOS API | Priority |
+| --- | --- | --- |
+| Parsing | ```(ACOAdaptiveCardParseResult*) fromJson: (NSString*)``` | P0 |
+| Rendering | ```(ACRRenderResult*) render:(ACOAdaptiveCard*) config:(ACOHostConfig) widthConstraint:(float)``` | P0 |
+| Rendering | ```(ACRRenderResult*) render:(ACOAdaptiveCard*) config:(ACOHostConfig) widthConstraint:(float) delegate:(id<ACRActionDelegate>)``` | P0 |
+| Rendering | ```(ACRRenderResult*) renderAsViewController:(ACOAdaptiveCard*) config:(ACOHostConfig) frame:(CGRect) delegate:(id<ACRActionDelegate>)``` | P0 |
+| Custom Element | ```(void) setBaseCardElementRenderer: (ACRBaseCardElementRenderer*)cardElementType: (ACRCardElementType)``` | P1 |
+| Custom Element | ```(void) setCustomElementParser: (NSObject<ACOIBaseCardElementParser>)``` | P1 |
+| Custom Element | ```(void) setCustomElementParser: (NSObject<ACOIBaseCardElementParser>) key: (NSString*)``` | P1 |
+| Custom Element | ```(void) setCustomElementRenderer: (ACRBaseCardElementRenderer*) key:(NSString *)``` | P1 |
+| Custom Element | ```(NSObject<ACOIBaseCardElementParser>*) getCustomElementParser: (NSString*)``` | P1 |
+| Custom Element | ```(ACRBaseCardElementRenderer*) getRenderer: (NSNumber *)``` | P1 |
+| Custom Action | ```(void) setActionRenderer: (ACRBaseActionElementRenderer) cardElementType: (NSNumber*)``` | P1 |
+| Custom Action | ```(ACRBaseActionElementRenderer*) getActionRenderer: (NSNumber*)``` | P1 |
+| Custom Action Set | ```(void) setActionSetRenderer: (id<ACRIBaseActionSetRenderer>)``` | P2 |
+| Custom Action Set | ```(id<ACRIBaseActionSetRenderer>) getActionSetRenderer``` | P2 | 
+| Resource Resolvers | ```(void) setResourceResolver: (NSObject<ACOIResourceResolver>) scheme: (NSString*)``` | P2 |
+| Resource Resolvers | ```(NSObject<ACOIResourceResolver>) getResourceResolverForScheme: (NSString*)``` | P2 |
+| Feature Registration | ```(void) addFeature: (NSString*) featureVersion: (NSString*)``` | P0 | 
+| Feature Registration | ```(NSString*) removeFeature: (NSString*)``` | P0 |
+| Feature Registration | ```(NSString*) getFeatureVersion: (NSString*)``` | P0 |
+| Resource Information | ```(NSArray<ACORemoteResourceInformation>*) remoteResourceInformation``` | P2 | 
+
+## Breaking changes
+
+There is one big breaking change to be introduced in the iOS platform due to the Xamarin bindings having a direct relationship with the name of the library, which requires changing the library name from ACFramework to AdaptiveCards.
+
+Besides that, no breaking changes should be introduced as the native platforms' code is only consumed and projected into C#.
+
+## Test Plan
```

Comment:
># Test Plan [](start = 1, length = 11)

[Team Spec Review] Add details around the new tool and the specific samples/scenarios we want to test here; REference these then in the last bug bash for November for collective testing; What about the unit testing story eventually? We should cover that